src/commands.rs

```diff
@@ -3160,12 +3182,48 @@ fn cmd_obslog(ui: &mut Ui, command: &CommandHelper, args: &ObslogArgs) -> Result
     } else {
         for commit in commits {
             template.format(&commit, formatter)?;
+            if let Some(diff_format) = diff_format {
+                show_predecessor_patch(formatter, &workspace_command, &commit, diff_format)?;
+            }
         }
     }
 
     Ok(())
 }
 
+fn show_predecessor_patch(
+    formatter: &mut dyn Formatter,
+    workspace_command: &WorkspaceCommandHelper,
+    commit: &Commit,
+    diff_format: DiffFormat,
+) -> Result<(), CommandError> {
+    let predecessors = commit.predecessors();
+    if !predecessors.is_empty() {
+        let predecessor = &predecessors[0];
+        let new_parent_tree =
+            merge_commit_trees(workspace_command.repo().as_repo_ref(), &commit.parents());
+        let parent_tree = if predecessor.parent_ids() == commit.parent_ids() {
+            new_parent_tree
+        } else {
+            // Rebase the predecessor to have the current commit's parent(s) and use that
+            // tree as base
+            let old_parent_tree = merge_commit_trees(
+                workspace_command.repo().as_repo_ref(),
+                &predecessor.parents(),
+            );
+            let rebased_tree_id =
+                merge_trees(&new_parent_tree, &old_parent_tree, &predecessor.tree())?;
```

Comment:
Good idea, but does this actually write to the repository? Should it only have the rebased commit in-memory instead? Otherwise we have no way to show the diff on a readonly repository (and in terms of API, you probably want this to be entirely client-side if possible, and/or to move the whole logic into the API and just get the actual diff back)